### PR TITLE
logseq: add darwin pkgs

### PR DESCRIPTION
**Pull request by klchen0112**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/logseq/linux.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{ lib
+, stdenv
+, fetchurl
+, appimageTools
+, makeWrapper
+  # graphs will not sync without matching upstream's major electron version
+, electron_24
```
Comment (kirillrdy) [inline pkgs/applications/misc/logseq/linux.nix:7]: Electron version 24.8.6 is EOL

Comment (klchen0112) [reply pkgs/applications/misc/logseq/linux.nix]: I am not very familiar with Electron. As mentioned in the link [release 0.9.19](https://github.com/logseq/logseq/blob/bd36efe4f7f5b56a59a160de75708270591ffc55/package.json#L113,) the Electron version for logseq 0.9.19 is still Electron 24. Therefore, it is necessary to use Electron 24. Alternatively,we can wait for the next release of logseq, where the Electron version in the master branch is already Electron 26.

Comment (kirillrdy) [reply pkgs/applications/misc/logseq/linux.nix:7]: from `nixpkgs` point of view electron_24 is marked as insecure due to it's EOL here are your options ( in no particular order )

 - you can try to see if it works with electron_25.
 -  wait for next release, 
 - cherry-pick enough patches from master to get it working with electron 26,
 - package unreleased commit, by marking version as "unstable" ( see contribution guidelines for more )

**Comment by kirillrdy** [comment]
@klchen0112 please resolve merge conflict

**Comment by klchen0112** [comment]
> @klchen0112 please resolve merge conflict

I resolved it. And all checks have passed.

**Comment by kirillrdy** [comment]
hi, so I just released ( was reviewing one of PRs) that there are 3 PRs releated to adding logseq darwin 

https://github.com/NixOS/nixpkgs/pull/268249
https://github.com/NixOS/nixpkgs/pull/248090
https://github.com/NixOS/nixpkgs/pull/191091

can I ask you to collaborate on a new PR ? feel free to reach out for help

looking forward on getting this merged !

**Comment by klchen0112** [comment]
> hi, so I just released ( was reviewing one of PRs) that there are 3 PRs releated to adding logseq darwin
> 
> #268249 #248090 #191091
> 
> can I ask you to collaborate on a new PR ? feel free to reach out for help
> 
> looking forward on getting this merged !

I'm willing to do this. But threre are some questions. 

1. It's necessary to use makeWrapper in darwin? [#268249](https://github.com/NixOS/nixpkgs/pull/268249)
2. For linux, I think this is enough.  [#248090](https://github.com/NixOS/nixpkgs/pull/248090) add `curl expat openssl zlib` in buildinputs
```nix
nativeBuildInputs = [ makeWrapper autoPatchelfHook ];
buildInputs = [ stdenv.cc.cc];
```
3. `runHook preInstall` and `runHook postInstall` should be added for darwin build?

**Comment by yzx9** [comment]
Closing as #331824 has been merged, which adds darwin support

**Pull request closed without merging**